src/core/server/elasticsearch/scoped_cluster_client.ts

```diff
@@ -72,10 +74,20 @@ export class ScopedClusterClient {
     clientParams: Record<string, unknown> = {},
     options?: CallAPIOptions
   ) {
-    if (this.headers !== undefined) {
-      clientParams.headers = this.headers;
-    }
+    const defaultHeaders = this.headers;
+    if (defaultHeaders !== undefined) {
+      const customHeaders: any = clientParams.headers;
+      if (isObject(customHeaders)) {
```

Comment:
if you already use lodash why don't use isPlainObject/isObject from it? at least we don't need to write tests

Reply:
lodash function does not type-guard the if statement so typescript errors out. Turns out if I define the `headers` into another variable instead of using 
object accessor immediately the type-guard is done implicitly.